### DISCUSSION TITLE
Problems getting 2FA working

### DISCUSSION THREAD

**Discussion by scsudo** [RakambdaOrg/ChannelPointsMiner#140] [Q&A] [answered]
Hello all,

unfortunately I have a problem starting the container in interactive mode to be able to do my 2FA authentication. It doesn't seem to find the config.json file despite mounting the appropriate config files and directories.

```
[twitchminer@localhost twitch-miner]# docker run -it --mount type=bind,source=/opt/twitch-miner/authentication,target=/usr/src/app/authentication --mount type=bind,source=/opt/twitch-miner/streamers,target=/usr/src/app/streamers --mount type=bind,source=/opt/twitch-miner/config.json,target=/usr/src/app/config.json mrcraftcod/channel-points-miner:latest
10:37:03,269 INFO [main] f.r.c.Main {} - Starting everything up (0.0.8 | ef7c133 | develop)
Exception in thread "main" java.lang.IllegalStateException: No main config found at /config.json
	at fr.raksrinana.channelpointsminer.factory.ConfigurationFactory.getInstance(ConfigurationFactory.java:31)
	at fr.raksrinana.channelpointsminer.Main.main(Main.java:42)
Caused by: java.nio.file.NoSuchFileException: config.json
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:380)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:432)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
	at java.base/java.nio.file.Files.newInputStream(Files.java:160)
	at fr.raksrinana.channelpointsminer.factory.ConfigurationFactory.getInstance(ConfigurationFactory.java:27)
	... 1 more
```

The config file is created and the path is correct.
```
[twitchminer@localhost twitch-miner]# find .
./docker-compose.yml
./streamers
./streamers/xxx.json
./authentication
./config.json
```

Does anyone know a solution?

**Comment by Rakambda** [comment, marked as answer]
You are missing the command parameter (hence the wrong path in the error `No main config found at /config.json`). If you look the [readme](https://github.com/RakSrinaNa/ChannelPointsMiner#docker) you'll see a `command: "--settings /usr/src/app/config.json"` in the compose file.

So I guess that for you it should end up being something like `docker run -it --mount type=bind,source=/opt/twitch-miner/authentication,target=/usr/src/app/authentication --mount type=bind,source=/opt/twitch-miner/streamers,target=/usr/src/app/streamers --mount type=bind,source=/opt/twitch-miner/config.json,target=/usr/src/app/config.json mrcraftcod/channel-points-miner:latest --settings /usr/src/app/config.json`

**Comment by scsudo** [comment]
Thank you for the quick reply. Been too dumb to read. Unfortunately, however, I then come to the next problem of the container exit directly after the start. I only get the Starting output
`10:57:33,413 INFO [main] f.r.c.Main {} - Starting everything up (0.0.8 | ef7c133 | develop)`
```
[twitchminer@localhost twitch-miner]# docker ps -a
ee54fec71988   mrcraftcod/channel-points-miner:latest         "java -cp @/app/jib-…"   19 seconds ago   Exited (0) 16 seconds ago                                                             jolly_jackson
```

**Reply by Rakambda** [reply]
I just realised the docker tags seems wrong 😆 latest contains the develop branch. But whatever, the develop works fine, I use it for quite a while.

Can you share the config.json content (obviously censoring the private data in it) ?

**Reply by scsudo** [reply]
Of course
```
{
  "accounts": [
    {
      "username": "xxx",
      "password": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "use2FA": true,
      "loadFollows": true,
      "enabled": false,
      "defaultStreamerSettings": {
        "makePredictions": false,
        "followRaid": true,
        "participateCampaigns": true
      },
      "streamerConfigDirectories": [
        {
          "path": "streamers",
          "recursive": false
        }
      ],
      "discord": {
        "webhookUrl": "https://discord.com/api/webhooks/xxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "embeds": true
      },
      "reloadEvery": 15
    }
  ]
}
```
Thanks for your help

**Reply by Rakambda** [reply]
Took me quite some time to get back a correct docker setup on my raspberry, all that to end up in the same case as you (was expected 😆 ). But maybe should have used my eyes first. You'll see it's quite stupid and we both had it in front of our eyes 😄

Now for the magic trick: `"enabled": false,` in the config does what is says. Basically the account is disabled and not currently mined.

**Reply by Rakambda** [reply]
I guess you took it from here https://github.com/RakSrinaNa/ChannelPointsMiner/blob/main/src/test/resources/config/config.json . Maybe I should update it to have it enabled, would be better I guess.

**Reply by scsudo** [reply]
OMG thank you 😂🙈. does not seem to be my day
